src/components/app/details/cicdHistory/TriggerDetails.tsx

```diff
@@ -0,0 +1,295 @@
+import React, { useState } from 'react'
+import { Progressing, showError, createGitCommitUrl, asyncWrap, ConfirmationDialog } from '../../../common'
+import { toast } from 'react-toastify'
+import { useRouteMatch, useLocation, useParams } from 'react-router'
+import { statusColor as colorMap } from '../../config'
+import { Moment12HourFormat } from '../../../../config'
+import moment from 'moment'
+import docker from '../../../../assets/icons/misc/docker.svg'
+import warn from '../../../../assets/icons/ic-warning.svg'
+import '../cIDetails/ciDetails.scss'
+import { CurrentStatusType, FinishedType, GitTriggers, HistoryComponentType, ProgressingStatusType, PROGRESSING_STATUS, StartDetailsType, TERMINAL_STATUS_COLOR_CLASS_MAP, TriggerDetailsType, WorkerStatusType } from '../cicdHistory/types'
+import { Link } from 'react-router-dom'
+import { cancelCiTrigger, cancelPrePostCdTrigger } from '../../service'
+
+export const TriggerDetails = React.memo(
+    ({
+        status,
+        startedOn,
+        finishedOn,
+        triggeredBy,
+        triggeredByEmail,
+        ciMaterials,
+        gitTriggers,
+        message,
+        podStatus,
+        type,
+        stage,
+        artifact,
+    }: TriggerDetailsType) => {
+        return (
+            <div
+                className="trigger-details"
+                style={{ height: '137px', display: 'grid', gridTemplateColumns: '60px 1fr' }}
+            >
+                <div className="trigger-details__status flex">
+                    <svg width="25" height="87" viewBox="0 0 25 87" fill="none" xmlns="http://www.w3.org/2000/svg">
+                        <circle cx="12.5" cy="6.5" r="6" fill="white" stroke="#3B444C" />
+                        <circle
+                            cx="12.5"
+                            cy="74.5"
+                            r="6"
+                            fill={colorMap[status?.toLowerCase()]}
+                            stroke={colorMap[status?.toLowerCase()]}
+                            strokeWidth="12"
+                            strokeOpacity="0.3"
+                        />
+                        <line x1="12.5" y1="11.9997" x2="12.5362" y2="69" stroke="#3B444C" />
+                    </svg>
```

Comment:
can we extract this SVG portion to a JSX FC?

Reply:
Done